frappe/public/js/frappe/widgets/number_card_widget.js

```diff
@@ -102,29 +160,42 @@ export default class NumberCardWidget extends Widget {
 	}
 
 	get_number() {
-		return frappe.xcall('frappe.desk.doctype.number_card.number_card.get_result', {
-			doc: this.card_doc
-		}).then(res => {
-			this.number = res;
-			if (this.card_doc.function !== 'Count') {
-				return frappe.model.with_doctype(this.card_doc.document_type, () => {
-					this.get_formatted_number();
-				});
+		return frappe.xcall(this.settings.method, this.settings.args).then(res => {
+			if (this.card_doc.type == 'Report') {
+				this.get_number_for_report(res);
 			} else {
-				this.number_html = res;
+				this.number = res;
+				if (this.card_doc.function !== 'Count') {
+					return frappe.model.with_doctype(this.card_doc.document_type, () => {
+						const based_on_df =
+							frappe.meta.get_docfield(this.card_doc.document_type, this.card_doc.aggregate_function_based_on);
+						this.get_formatted_number(based_on_df);
+					});
+				} else {
+					this.number_html = res;
```

Comment:
Instead of the server sending HTML, can we have something like used in report summary?

https://github.com/frappe/frappe/blob/9fc93a80f24fca1e56e0771bde0d1944dae6a718/frappe/public/js/frappe/widgets/utils.js#L102-L118

We can add more data type cases to this function.